WalletWasabi/Helpers/IoHelpers.cs

```diff
@@ -11,49 +11,57 @@ namespace System.IO
 {
 	public static class IoHelpers
 	{
-		// http://stackoverflow.com/a/14933880/2061103
-		public static async Task DeleteRecursivelyWithMagicDustAsync(string destinationDir)
+		/// <summary>
+		/// Attempts to delete <paramref name="directory"/> with retry feature to allow File Explorer or any other
+		/// application that holds the directory handle of the <paramref name="directory"/> to release it.
+		/// </summary>
+		/// <see href="https://stackoverflow.com/questions/329355/cannot-delete-directory-with-directory-deletepath-true/14933880#44324346"/>
+		public static async Task<bool> TryDeleteDirectoryAsync(string directory, int maxRetries = 10, int millisecondsDelay = 100)
 		{
-			const int MagicDust = 10;
-			for (var gnomes = 1; gnomes <= MagicDust; gnomes++)
+			if (directory is null)
+			{
+				throw new ArgumentNullException(directory);
+			}
+
+			if (maxRetries < 1)
+			{
+				throw new ArgumentOutOfRangeException(nameof(maxRetries));
+			}
+
+			if (millisecondsDelay < 1)
+			{
+				throw new ArgumentOutOfRangeException(nameof(millisecondsDelay));
+			}
+
+			for (int i = 0; i < maxRetries; ++i)
 			{
 				try
 				{
-					Directory.Delete(destinationDir, true);
+					if (Directory.Exists(directory))
+					{
+						Directory.Delete(directory, recursive: true);
+					}
+
+					return true;
 				}
-				catch (DirectoryNotFoundException)
+				catch (DirectoryNotFoundException e)
 				{
-					return; // good!
+					Logger.LogDebug($"Directory was not found: {e}");
+
+					// Directory does not exist. So the operation is trivially done.
+					return true;
 				}
 				catch (IOException)
```

Comment:
Use `when` to catch multiple types of exceptions and remove the second Task.Delay

Reply:
Done.